code/modules/mob/living/carbon/human/species_types/golems.dm

```diff
@@ -489,3 +489,93 @@
 		H.adjustFireLoss(-4)
 		H.reagents.remove_reagent(chem.id, REAGENTS_METABOLISM)
 
+/datum/species/golem/cloth
+	name = "Cloth Golem"
+	id = "cloth"
+	limbs_id = "clothgolem"
+	sexes = FALSE
+	info_text = "As a <span class='danger'>Cloth Golem</span>, you are able to reform yourself after death, provided your remains aren't burned or destroyed. You are, of course, very flammable."
+	species_traits = list(NOBREATH,RESISTCOLD,RESISTPRESSURE,NOGUNS,NOBLOOD,RADIMMUNE,VIRUSIMMUNE,PIERCEIMMUNE,NODISMEMBER) //no mutcolors, and can burn
+	armor = 15 //feels no pain, but not too resistant
+	burnmod = 2 // don't get burned
+	speedmod = 1 // not as heavy as stone
+	punchdamagelow = 4
+	punchstunthreshold = 7
+	punchdamagehigh = 8 // not as heavy as stone
+
+/datum/species/golem/cloth/spec_life(mob/living/carbon/human/H)
+	if(H.fire_stacks < 1)

```

Comment:
This is intentional so they don't stop burning naturally if ignited, but the message isn't. I guess it does give players a warning about their weakness, so it could be fine.